REDCapUITweaker.php

```diff
@@ -2104,7 +2144,58 @@ function validateSettings( $settings )
 		return null;
 	}
 
-
-
+        function provideSQLCheckBox( $listFields,$project_id, $record, $event_id, $instrument, $instance)
+        {
+            global $Proj;
+            foreach ($listFields as $infoField )
+            {
+                $fieldname = $infoField['field_name']; 
+                $sqlfieldname = \Form::getValueInQuotesActionTag($infoField['field_annotation'], "@SQLCHECKBOX");
+                try 
+                {
+                    if(!isset($Proj->metadata[$sqlfieldname]) || $Proj->metadata[$sqlfieldname]['element_type'] !== 'sql')
+                    {
+                        throw new \Exception('SQL field ('.$sqlfieldname.') does not exist.');
+                    }
+                    $SQLValue = $Proj->metadata[$sqlfieldname]['element_enum'];
+                    if($SQLValue == '')
+                    {
+                        throw new \Exception('SQL for field ('.$sqlfieldname.') is not defined.');
+                    }
+                    $enum = getSqlFieldEnum($SQLValue, $project_id, $record, $event_id, $instance, null, null, $instrument);
+
+                    $Locking = new \Locking();
+                    $Locking->findLocked($Proj, $record, $fieldname, $event_id);
+                    $Locking->findLockedWholeRecord($project_id, $record);
+                    $locked = !empty($Locking->lockedWhole) || !empty($Locking->locked[$record][$event_id][$instance]);
+
+                    if (!$locked && $this->isModuleEnabled('locking_forms')) 
+                    {
+                        $Locking = \ExternalModules\ExternalModules::getModuleInstance('locking_forms');
+                        $locked = $Locking->isHardLocked($event_id, $instrument);
+                    }
+
+                    if ($Proj->metadata[$fieldname]['element_enum'] !== $enum && !$locked)
+                    {
+                        $Proj->metadata[$fieldname]['element_enum'] = $enum;
+
+                        $sql = "UPDATE redcap.redcap_metadata SET element_enum = ? WHERE project_id = ? AND field_name = ?";
+                        if ($this->query($sql, [$enum, $project_id, $fieldname])) {
+                            if ($enum == "") {
+                                $enum = "No options returned";
+                            }
+                            \REDCap::logEvent('UI Tweaker-@SQLCHECKBOX', "Update check box options\nField=" . $fieldname . "\nOptions=" . $enum, "UI Tweaker", $record, "", $project_id);
```

Comment:
I would suggest referring to the module by its full name *REDCap UI Tweaker* in the logs.

Reply:
Updated